src/app/clusters/content-launch-server/content-launch-delegate.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/af-structs.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/enums.h>
+
+#include <app/util/af.h>
+#include <lib/core/Optional.h>
+#include <list>
+
+struct ContentLaunchResponse
+{
+    CHIP_ERROR err;
+    chip::CharSpan data;
+    chip::app::Clusters::ContentLauncher::ContentLaunchStatus status;
+};
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ContentLauncher {
+
+/** @brief
+ *    Defines methods for implementing application-specific logic for the Content Launcher Cluster.
+ */
+class Delegate
+{
+public:
+    virtual ContentLaunchResponse HandleLaunchContent(chip::EndpointId endpointId, std::list<ContentLaunchParamater> parameterList,
```

Comment:
Is passing the list by value there right?  I would have expected passing by const reference.

Reply:
Updated.